src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -123,6 +142,129 @@ private static BidRequest fillExplicitParameters(BidRequest bidRequest, RoutingC
                 : bidRequest;
     }
 
+    private BidRequest overwriteParameters(BidRequest bidRequest, HttpServerRequest request) {
+        final List<Format> overwrittenFormats = overwriteBannerFormats(request);
+        final Site updatedSite = overwriteSitePage(bidRequest.getSite(), request);
+        final String tagId = request.getParam(SLOT_REQUEST_PARAM);
+
+        final Imp updatedImp = updateImp(bidRequest.getImp().get(0), overwrittenFormats, tagId);
+
+        final Integer timeout = parseIntParam(request, TIMEOUT_REQUEST_PARAM);
+        final Integer updatedTimeout = timeout != null ? timeout - timeoutAdjustmentMs : null;
+
+        return updateBidRequest(bidRequest, updatedSite, updatedImp, updatedTimeout);
+    }
+
+    private static List<Format> overwriteBannerFormats(HttpServerRequest request) {
+        final Integer ow = parseIntParam(request, OW_REQUEST_PARAM);
```

Comment:
The code can be simplified a little:
```
        final Integer ow = parseIntParam(request, OW_REQUEST_PARAM);
        final Integer formatWidth = ow != null ? ow : parseIntParam(request, W_REQUEST_PARAM);

        final Integer oh = parseIntParam(request, OH_REQUEST_PARAM);
        final Integer formatHeight = oh != null ? oh : parseIntParam(request, H_REQUEST_PARAM);

        final Format format = formatWidth != null || formatHeight != null
                ? Format.builder().w(formatWidth).h(formatHeight).build()
                : null;

        final String ms = request.getParam(MS_REQUEST_PARAM);
        final List<Format> multiSizeFormats = StringUtils.isNotBlank(ms) ? parseMultiSizeParam(ms) : null;
```
So, the computing of `w` and `h` vars can be omitted if `ow` or `oh` existed.